pkg/rebuild/rebuild/storage.go

```diff
@@ -102,15 +103,25 @@ func AssetCopy(ctx context.Context, to, from AssetStore, a Asset) error {
 	if _, err := io.Copy(w, r); err != nil {
 		return errors.Wrap(err, "copy failed")
 	}
-	return nil
+	return w.Close()
 }
 
 // DebugStoreFromContext constructs a DebugStorer using values from the given context.
 func DebugStoreFromContext(ctx context.Context) (AssetStore, error) {
 	if uploadpath, ok := ctx.Value(UploadArtifactsPathID).(string); ok {
-		if strings.HasPrefix(uploadpath, "gs://") {
+		u, err := url.Parse(uploadpath)
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing UploadArtifactsPathID as url")
+		}
+		if u.Scheme == "gs" {
 			storer, err := NewGCSStore(ctx, uploadpath)
 			return storer, errors.Wrapf(err, "Failed to create GCS storer")
+		} else if u.Scheme == "file" {
+			path, _ := strings.CutPrefix(uploadpath, "file://")
```

Comment:
can you use u.Path?

Reply:
I thought I couldn't because I was doing "file://a/b/c" which cuts "a" as the host and "/b/c" as the path, but reading the spec I think the format I was using is incorrect. Instead I should have a leading slash on the path section for three slashes in a row: "file:///a/b/c" which should work.